core/unit_test/TestHalfConversion.hpp

```diff
@@ -20,11 +20,13 @@ namespace Test {
 
 template <class T>
 void test_half_conversion_type() {
-  double epsilon                 = KOKKOS_HALF_T_IS_FLOAT ? 0.0000003 : 0.0003;
+  // When truncating mantissa to 10bits (like f16), 3.3 becomes 3.298828125
+  // 3.3 - 3.298828125 < 1.1719e-3, so conversion error should be smaller
+  double epsilon                 = KOKKOS_HALF_T_IS_FLOAT ? 3e-7 : 1.1719e-3;
```

Comment:
I don't quite understand where that failed.
What architecture did you run on?  How did you configure Kokkos?
This is relaxing the tolerance by 10x, I would like to understand more before we merge.

Reply:
By hand, supposing that `T` is higher precision than fp16, the cast will just truncate the mantissa to 10 bits, so the absolute truncation error is less than 1 ulp. For fp16, the error is `abs_err = | T(3.3) - fp16(T(3.3)| =  exp/(2**10)` with `exp=2` for 3.3, so `abs_err = 0.001953125`. My bound is not as tight as Paul's because the truncation error is less than 1 ulp.

Reply:
Thx